coopr-server/src/main/java/co/cask/coopr/spec/template/LeaseDuration.java

```diff
@@ -38,6 +40,50 @@ public LeaseDuration(long initial, long max, long step) {
     this.step = step;
   }
 
+  public LeaseDuration(String initial, String max, String step) {
+    this(getTimestamp(initial), getTimestamp(max), getTimestamp(step));
+  }
+
+  /**
+   * Returns a timestamp in milliseconds.
+   *
+   * @param base The string argument user provided.
+   * @return Timestamp in milliseconds
+   */
+  private static long getTimestamp(String base) {
+    if (base.equals("min") || base.equals("0")) {
+      return 0L;
+    }
+
+    if (base.equals("max")) {
+      return Long.MAX_VALUE;
+    }
+
+    if (base.matches("^[0-9]+$")) {
+      // only numbers, no time unit: millisecond format
+      return Long.parseLong(base);
+    }
+
+    try {
+      char type = base.charAt(base.length() - 1);
+      int offset = Integer.parseInt(base.substring(0, base.length() - 1));
+      switch (type) {
+        case 's':
+          return TimeUnit.SECONDS.toMillis(offset);
+        case 'm':
+          return TimeUnit.MINUTES.toMillis(offset);
+        case 'h':
+          return TimeUnit.HOURS.toMillis(offset);
+        case 'd':
+          return TimeUnit.DAYS.toMillis(offset);
+        default:
+          throw new RuntimeException("Unsupported relative time format: " + type);
+      }
+    } catch (NumberFormatException e) {
+      throw new RuntimeException("Invalid number value: " + base + ". Reason: " + e.getMessage());
```

Comment:
Do the http handlers catch this and return a 400?  Don't remember, but they might only look for IllegalArgumentExceptions so we might want to change to that.